src/main.cc

```diff
@@ -178,6 +178,39 @@ signal_cb(ev_loop *loop, struct ev_signal *w, int revents)
 	tarantool_exit(0);
 }
 
+static void
+signal_sigint_cb(ev_loop *loop, struct ev_signal *w, int revents)
+{
+	(void) loop;
+	(void) w;
+	(void) revents;
+
+	/**
+	 * If running in daemon mode, tarantool exits on SIGINT.
+	 * Complain about possibly sudden and unexpected death.
+	 */
+	if (pid_file) {
+		say_crit("got signal %d - %s", w->signum, strsignal(w->signum));
+		tarantool_exit(0);
+	}
```

Comment:
This case is not covered in tests. The patch adds a SIGINT handler, but test covers sending INT signal in interactive mode only. I would add a test with sending INT signal when Tarantool is a daemon mode too.
Something like this:
- start Tarantool as a daemon: `TT_PID_FILE=tarantool.pid TT_LOG=tarantool.log TT_BACKGROUND=true TT_LISTEN=3301 tarantool -e 'box.cfg{}'`
- send INT signal to a Tarantool process
- make sure process terminated
- make sure appropriate message found in tarantool.log

Reply:
Thank you for idea! The test is added.